lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -669,12 +671,34 @@ private void handleStateRequest() {
 		requestedState = null;
 	}
 
+	private boolean checkConnectionStale() {
+	    if(lastActivity == 0) {
+	        return false;
+        }
+		long now = System.currentTimeMillis();
+		long intervalSinceLastActivity = now - lastActivity;
+		if(intervalSinceLastActivity > (maxIdleInterval + connectionStateTtl)) {
+			/* RTN15g1, RTN15g2 Force a new connection if the previous one is stale */
+			if(connection.id != null) {
+				connection.id = null;
```

Comment:
I think we should add a log message here about why the connection state is explicitly being cleared

Reply:
@mattheworiordan done https://github.com/ably/ably-java/pull/390/commits/0a1c622ce8f806181ffb3207d980608ab7388e82